### PR TITLE
Add branch name to pihole -v

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Like so:

```
root@raspberrypi:/etc/.pihole# pihole -v
  Pi-hole version is v4.3.5-460-g413fa94 (Latest: v4.4) Branch: release/v5.0
  AdminLTE version is v4.3.2-462-g8e51f754 (Latest: v4.3.3) Branch: release/v5.0
  FTL version is vDev-853261b (Latest: v4.3.1) Branch: release/v5.0
```

### CODE REVIEW

**Comment by dschaper** [comment]
Do we need to output if the branch is master? 

Would this look better?
`Pi-hole version is v4.3.5-460-g413fa94 release/v5.0 (Latest: v4.4)`
`Pi-hole version is v4.3.5 master (Latest: v4.4)`

I think there's a few things in the new function that could be tweaked, I'll push some bash stuff in a bit.

**Comment by PromoFaux** [comment]
Funnily enough i did it that way around the first time. Then changed my mind, but on reflection I think it looks better to have it directly after the version.

Unsure about `master` or not...

**Comment by dschaper** [comment]
Trust your instincts... 

I'm half and half with master. Dom can break the tie.

**Comment by PromoFaux** [comment]
On reflection, I think branch first, then versions:

```
Pi-hole version is release/v5.0 v4.3.5-460-g413fa94 (Latest: v4.4)
AdminLTE version is release/v5.0 v4.3.2-462-g8e51f754 (Latest: v4.3.3)
FTL version is release/v5.0 vDev-853261b (Latest: v4.3.1)
```

File: advanced/Scripts/version.sh [hunk 1]
```diff
@@ -97,22 +112,48 @@ getRemoteVersion(){
...
 }
 
+getLocalBranch(){
+    # Get the checked out branch of the local directory
+    local directory="${1}"
+    local branch
+
+     # Local FTL btranch is stored in /etc/pihole/ftlbranch
+    if [[ "$1" == "FTL" ]]; then
+        branch=$(cat /etc/pihole/ftlbranch)
```
Comment (DL6ER) [inline advanced/Scripts/version.sh:122]: This file is not available if the user has never done a local checkout of a custom FTL branch, the output would be
```plain
cat: /etc/pihole/ftlbranch: No such file or directory
  Current FTL version is  v4.3.1-800-g853261bb.
```

Use
```suggestion
        branch="$(pihole-FTL branch)"
```
to get the branch of the installed FTL instance reliably (`pihole-FTL -b` is the short form but it doesn't hurt to be explicit).

File: advanced/Scripts/version.sh [hunk 2]
```diff
@@ -97,22 +112,48 @@ getRemoteVersion(){
...
+        if [[ "${branch}" == "master" ]]; then
+            echo ""
+        else
+            echo "${branch} "
+        fi
+    elif [[ "${branch}" == "${DEFAULT}" ]]; then
+        echo "ERROR"
+        return 1
+    else
+        echo "No Checked out branch"
```
Comment (DL6ER) [inline advanced/Scripts/version.sh:137]: Could this result in
```
Pi-hole version is No Checked out branchv4.3.5-460-g413fa94 (Latest: v4.4)
```
(also note the missing space between branch and v4.3.5)?

I'm not sure how/if we could reach this `else` condition.

File: advanced/Scripts/version.sh (hunk 1 above)
Comment (DL6ER) [reply advanced/Scripts/version.sh]: I pushed this change already, feel free to resolve this conversation if you agree.

**Comment by DL6ER** [comment]
My last commit will handle detached HEAD mode as well. Example:
```
$ cd /var/www/html/admin

$ git checkout 322d9207 # <------ randomly selected commit hash
Note: checking out '322d9207'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 322d9207 Run phpcs to have Pi-hole's PHP code match the PSR-12 standard.                                                                      

$ pihole -v -a
  Current AdminLTE version is in detached HEAD state at v4.3.2-327-g322d9207.

File: advanced/Scripts/version.sh (hunk 2 above)
Comment (PromoFaux) [reply advanced/Scripts/version.sh:139]: We can probably lose the `else` section... the function was a copy-paste, and then tweak of `getLocalVersion()`

**Comment by DL6ER** [comment]
This will print
```
$ pihole -v -a
  Current AdminLTE version is release v4.3.2.
```
if the branch comes out as `v*` (even though I don't think this would happen).

**Pull request merged**